### ISSUE TITLE
makeEWSRequestAsync() fails with error 401 in Outlook clients intermittently.

### ISSUE THREAD

**Issue by glr0221** [OfficeDev/office-js#3205]
> Message from office-js bot:
We’re closing this issue because it has been inactive for a long time. We’re doing this to keep the issues list manageable and useful for everyone. If this issue is still relevant for you, please create a new issue.
> Thank you for your understanding and continued feedback.

# Provide required information needed to triage your issue
My sincerest apologies if this is not the right venue to submit this issue. We just have no other choice but to ask for help from MS.

## Your Environment

* Platform [PC desktop, Mac, iOS, Office on the web]: PC Windows with Exchange on premise.
* Host [Excel, Word, PowerPoint, etc.]: Outlook
* Office version number: 16.0.5257.1000 
* Operating System: Windows
* Browser (if using Office on the web): Chrome

## Expected behavior

Calling makeEWSRequestAsync() will reliably succeed in authenticating and executing the EWS-SOAP request.

## Current behavior

We currently have an Outlook add-in that uses EWS-APIs and works correctly when our customers use it on OWA. 

When they use the Outlook client however, it intermittently fails with an HTTP-401 error. [When we checked the network packets, we saw an HTTP 401 for our the GetItem() calls.](https://imgur.com/YM3T94c)

In some cases, when our users close and restart outlook, then the addin works perfectly. [Upon investigation, we found out that there was a successful call to GetClientAccessToken()](https://imgur.com/0jkAwO3). 

To avoid issues with authentication, how can we setup/invoke/configure makeEWSRequestAsync() to call GetClientAccessToken() all the time? Or is [getCallbackTokenAsync()](https://learn.microsoft.com/en-us/javascript/api/outlook/office.mailbox?view=outlook-js-1.5&preserve-view=true#outlook-office-mailbox-getcallbacktokenasync-member(2)](https://learn.microsoft.com/en-us/javascript/api/outlook/office.mailbox?view=outlook-js-1.5&preserve-view=true#outlook-office-mailbox-getcallbacktokenasync-member(2))) a recommended way to forcibly request for a token in an Exchange-on-premise environment? 

## Steps to reproduce

1. Using the right outlook client version and Exchange on-premise, invoke makeEWSRequestAsync to GetItem() using an addin.
2. Intermittently, makeEWSRequestAsync will fail with HTTP-401 error (Unauthorized).
3. Restart outlook and retry step 1, this time it will work.

## Link to live example(s)

1. ______
2. ______
3. ______

# Provide additional details

1. ______
2. ______
3. ______

## Useful logs

- [ ] Console errors
- [ ] Screenshots
- [ ] Test file (if only happens on a particular file)

Thank you for taking the time to report an issue. Our triage team will respond to you in less than 72 hours. Normally, response time is <10 hours Monday through Friday. We do not triage on weekends.

**Comment by alison-mk** [comment]
Hi @glr0221, thank you for submitting this issue! I've assigned it to the engineering team who will be able to provide you with more information or help resolve the issue. 

Cheers,
Alison

**Comment by glr0221** [comment]
Just a quick update, I found out that makeEWSRequestAsync() uses an expired token after some time. This is to be expected though, if the addin was not closed. However, even if the addin was closed and re-launched makeEWSRequestAsync() still uses the cached token. Is this by design? Or is there some configuration in Outlook or Exchange that we need to tweak? Thank you.

Is there any way to force makeEWSRequestAsync() to request for a new token? Or would calling getCallbackTokenAsync() just before makeEWSRequestAsync() achieve this forced token request? Thanks.

**Comment by timwan10** [comment]
So what SHOULD happen is this: 

1) Addin Calls MakeEWSRequestAsync
2) Behind the scenes Outlook calls GetClientAccessToken and stores it in an in memory cache.
3) Outlook Makes EWS Call with token from step 2 

4) Add-in makes Another makeewsrequest call
5) Outlook sees token from step 2 is still valid 
6) Make EWS Call with token from  step 2

7) Add-in makes another MakeEwsRequest CAll
8) The token is no longer valid. (the TTL is expired) 
9) Outlook makes another GetClientAccessToken call, to get a new token and stores it in memory. 
10) Outlook makes EWS Call with Token from step 9

If the Token in step 2 has expired. Then there should automatically be another request for a fresh token. (There is actually some leeway, Outlook will still go fetch a token as soon as there is ~10 minutes left...or something like that)

When you took the fiddler for GetClientAccessToken, the Time to Live (TTL) should have been in the XML response. (i.e. if this is 60 minutes... then Outlook will get a new token after 50 minutes has passed). 

Are you seeing no new token fetched AFTER 50 minutes? or are you seeing the token stop working BEFORE 50 minutes? (replace 50 with whatever the TTL time was in the GetClientAccessToken Response). 

Calling getCallbackTokenAsync()  should have no effect on this.

**Comment by glr0221** [comment]
Thank you @timwan10 .

After your notes, I took some time to check the 'exp' data of the JWT passed by the addin's makeEWSRequestAsync() call. And it turns out that the used token's expiration was dated 'Wed Mar 01 2023 14:56:06 GMT'. But it was used on a request dated Fri, 03 Mar 2023 23:09:45 GMT. 

What could possibly cause the addin to use an expired token when calling makeEWSRequestAsync()? Is there some setting in the Exchange server that can affect his behavior? Or is it because the outlook client remained open for a long time? The weird thing here though is that the addin was opened and closed at least 3 times. And each time, the same expired token was being used. 

It seems that it is not calling step #2. However, when outlook was closed and the addin was re-launched, everything was working fine. I have some network captures but will not be able to share them due to confidentiality. 

What can we do programmatically to force the makeEWSRequestAsync() to request a new token each time it launches? Or what server config do we have to tweak to correct this behavior?

Thank you so very much.
